core/domain/story_domain.py

```diff
@@ -775,7 +775,10 @@ def validate(self):
         if not isinstance(self.notes, str):
             raise utils.ValidationError(
                 'Expected notes to be a string, received %s' % self.notes)
-
+        if len(self.notes) > 5000:
+            raise utils.ValidationError(
+                'Expected notes length to be less than or equal to 5000,'
+                ' but it is %s long' % len(self.notes))
```

Comment:
Is it also possible to validate this in the handler schema?

Reply:
Are you sure? Aren't the nodes part of some other object that is included in some handler?

Reply:
There are some handlers which are using nodes but this is the validation of story notes. I have checked the files of story_editor.py and story_viewer.py which are the only files imported related to story in main.py and both of these files are not using notes. Am I wrong at some point ?

Reply:
There is a validation in frontend where it restricts story notes not to exceed 4000 characters and incase if it crosses 4000, then it will disable the save button so that the user can't save the story notes 
![image](https://user-images.githubusercontent.com/78922463/166802149-d43ea35e-a0ca-47d1-b210-83a93113b1b4.png)

Reply:
But that is weird as we check for 5000 but in frontend we check for 4000. @lkbhitesh07 What should be the correct value?

Reply:
Hey @vojtechjelinek, Sorry for the late reply I somehow missed this comment, Actually according to the spreadsheet [here](https://docs.google.com/spreadsheets/d/1kNLF-UkS_Qk0igNCGwLjJXM0ByHalqli41GPvWF43PE/edit#gid=107631643), it should be `5000` but I worked upon this frontend validation and when I checked with `5000` characters then it seemed to be too much so I had a discussion with Eric and we decided to keep it to `4000`, I will update the spreadsheet.
Thanks

Reply:
Yes

Reply:
I have changed the backend validation to 4000 characters. @vojtechjelinek PTAL

Reply:
Should we rerun the job too?

Reply:
I guess yes, because We checked whether the length is >5000 or not but since we changed the number of characters, we have to rerun the job now. So, should I add those files again and submit the testing form ?